jetty-integrations/jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdConfiguration.java

```diff
@@ -38,6 +38,165 @@
  */
 public class OpenIdConfiguration extends ContainerLifeCycle
 {
+/**
+ * <p>Builder for {@link OpenIdConfiguration}.</p>
+ * <p>The only required parameters are {@link #issuer}, {@link #clientId} and {@link #clientSecret}.</p>
+ * <p>The {@link #authorizationEndpoint}, {@link #tokenEndpoint} and {@link #endSessionEndpoint}  will be retrieved
+ * from the OpenID metadata at .well-known/openid-configuration if they are not explicitly specified.</p>
+ */
+    public static class Builder
+    {
+        private String issuer;
+        private String clientId;
+        private String clientSecret;
+
+        private String authorizationEndpoint;
+        private String tokenEndpoint;
+        private String endSessionEndpoint;
+        private String authenticationMethod;
+        private HttpClient httpClient;
+
+        private boolean authenticateNewUsers = false;
+        private boolean logoutWhenIdTokenIsExpired = false;
+        private final List<String> scopes = new ArrayList<>();
+
+        /**
+         * Create a Builder for an OpenID Configuration.
+         */
+        public Builder()
+        {
+        }
```

Comment:
Either:
```suggestion
        {
            this(null, null, null);
        }
```
Or see below: